pkg/locking/locking.go

```diff
@@ -219,7 +219,7 @@ func Initialize(coreObj *core.Core, opts ...InitializeOpt) (*core.ControlSession
 		break
 	}
 	if err != nil {
-		return nil, nil, fmt.Errorf("all authentications failed")
+		return nil, nil, fmt.Errorf("all authentications failed: %w", err)
```

Comment:
@yogev-lb : this is a bit misleading in the general case, as the `err` that will get reported up the call chain is just the _last_ of the failures that happened, and it's not guaranteed that all `ic.auths` failed with the same error. on the other hand, i suspect that most actual use cases will have just the one method set, so it'll do the right thing in the common case.

how about:
```suggestion
		return nil, nil, fmt.Errorf("all authentications failed, last with: %w", err)
```
that will make it both true and useful. :)